pytest_timeout.py

```diff
@@ -433,6 +432,43 @@ def dump_stacks():
         write("".join(traceback.format_stack(frame)))
 
 
+def _getdimensions():
+
+    if sys.version_info >= (3, 3, 0):
+        import shutil
+
+        size = shutil.get_terminal_size()
+        return size.lines, size.columns
+    else:
+        import termios, fcntl, struct
+
+        call = fcntl.ioctl(1, termios.TIOCGWINSZ, "\000" * 8)
+        height, width = struct.unpack("hhhh", call)[:2]
+        return height, width
+
+
+def _get_terminal_width():
+    width = 0
+    try:
+        _, width = _getdimensions()
+    except (KeyboardInterrupt, SystemExit, MemoryError, GeneratorExit):
+        raise
+    except:
+        # pass to fallback below
+        pass
+
+    if width == 0:
```

Comment:
Why have a sentinel of 0?  You can just directly put the block of this if condition in the except block which I think is better style.